src/app/main/connection-window/connection-window.component.ts

```diff
@@ -54,6 +55,17 @@ export class ConnectionWindowComponent extends SubscriptionHandler implements On
     }
   }
 
+  /*
+    AfterViewInit, a lifecycle hook that is called after Angular has fully initialized a component's view.
+    forces the login if auto_login is detected.
+  */
+  ngAfterViewInit(): void {
+    const auto_login = localStorage.getItem('auto_login') === 'true';
+    if (auto_login) {
+      this.myDiv.nativeElement.click();
```

Comment:
I don't think this is a good strategy. Why doing all of this in order to click that button that basically only triggers `onConnect()` while we should. just call `onConnect()` from here? :)

Reply:
How else would we remember if the user has logged in before. This is to auto login, that was the behaviour requested.

Reply:
@willkillson yes I understand but my point was different. The button triggers `onConnect()` when clicked.

So my point was: let's just call `this.onConnect()` to do the auto login instead of clicking the button, there is no reason to click the button when you could just call that method from there:)

Reply:
```
if (auto_login) {     
  this.onConnect();
...
```

Reply:
Oh, that makes sense.